iam/snippets/snippets.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	iam "google.golang.org/api/iam/v1"
+)
+
+var client, _ = google.DefaultClient(
+	context.Background(),
+	iam.CloudPlatformScope)
+var iamService, _ = iam.New(client)
+
+// [START iam_view_grantable_roles]
+func viewGrantableRoles(fullResourceName string) {
```

Comment:
We usually pass the project ID as an argument. It's a bit easier to test and it shows more clearly how to construct the full resource name.

Reply:
What if the resource isn't a project?

Reply:
If that's part of the documentation, sounds good.

Reply:
Yes, the docs explain what fullResourceName should be.